ui/src/components/category/CategoryForm.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { useForm } from "react-hook-form";
+import {Box, Button, TextField, MenuItem, Select, FormControl, FormLabel, FormHelperText } from "@mui/material";
+import { useState, useEffect } from "react";
+
+interface CategoryFormData {
+    name: string;
+    currency: string;
+}
+
+export default function CategoryForm({ onCategoryAddedAction }: { onCategoryAddedAction: () => void }) {
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+        reset,
+    } = useForm<CategoryFormData>();
+
+    const [userEmail, setUserEmail] = useState("");
+    const [currency, setCurrency] = useState("EUR");
+
+    useEffect(() => {
+        const emailCookie = document.cookie
```

Comment:
Read about retrieving cookies here: https://stackoverflow.com/questions/76595420/how-to-use-cookies-in-client-components-in-next-js-13-and-read-them-on-initial-s, first answer should sufficient to implement. YOu need to create wrapper component of server side.